### PR TITLE
fix broken Mtr chart

### LINKED ISSUE #531: Broken MTR chart
https://policyengine.org/us/household?focus=householdOutput.mtr&household=30669

image

Also happened yesterday with a different reform

### PR DESCRIPTION
**Pull request by sumitshinde-84**
### Summary
🔧🎨🆕

This pull request enhances the policy simulation interface by improving the variable editor and the marginal tax rates components. It fixes bugs, optimizes performance, adapts to data changes, and adds new features. It also improves the code quality and style of `src/pages/household/input/VariableEditor.jsx` and `src/pages/household/output/MarginalTaxRates.jsx`.

> _Sing, O Muse, of the skillful heroes who refined the code_
> _Of the variable editor, the splendid tool of simulation,_
> _Who fixed the linting error, the bane of the compiler,_
> _And optimized the URL, the swift messenger of data._

### Walkthrough
*  Move and refactor `MarginalTaxRates` component to use in variable editor page ([link](https://github.com/PolicyEngine/policyengine-app/pull/543/files?diff=unified&w=0#diff-43f598ac8f8928c8be5adb376f0f5e8d25a5f61fb6dea523f1157c84d2517ed9L1-R5), [link](https://github.com/PolicyEngine/policyengine-app/pull/543/files?diff=unified&w=0#diff-43f598ac8f8928c8be5adb376f0f5e8d25a5f61fb6dea523f1157c84d2517ed9L20-R25), [link](https://github.com/PolicyEngine/policyengine-app/pull/543/files?diff=unified&w=0#diff-43f598ac8f8928c8be5adb376f0f5e8d25a5f61fb6dea523f1157c84d2517ed9R38), [link](https://github.com/PolicyEngine/policyengine-app/pull/543/files?diff=unified&w=0#diff-43f598ac8f8928c8be5adb376f0f5e8d25a5f61fb6dea523f1157c84d2517ed9L55-R59), [link](https://github.com/PolicyEngine/policyengine-app/pull/543/files?diff=unified&w=0#diff-43f598ac8f8928c8be5adb376f0f5e8d25a5f61fb6dea523f1157c84d2517ed9R71), [link](https://github.com/PolicyEngine/policyengine-app/pull/543/files?diff=unified&w=0#diff-43f598ac8f8928c8be5adb376f0f5e8d25a5f61fb6dea523f1157c84d2517ed9R86), [link](https://github.com/PolicyEngine/policyengine-app/pull/543/files?diff=unified&w=0#diff-43f598ac8f8928c8be5adb376f0f5e8d25a5f61fb6dea523f1157c84d2517ed9L91-R217))
  - Initialize state variables as empty arrays to avoid errors ([link](https://github.com/PolicyEngine/policyengine-app/pull/543/files?diff=unified&w=0#diff-43f598ac8f8928c8be5adb376f0f5e8d25a5f61fb6dea523f1157c84d2517ed9L20-R25))
  - Disable linting rule for unused dependencies in `useEffect` hook ([link](https://github.com/PolicyEngine/policyengine-app/pull/543/files?diff=unified&w=0#diff-43f598ac8f8928c8be5adb376f0f5e8d25a5f61fb6dea523f1157c84d2517ed9R38))
  - Format code and split long line for readability ([link](https://github.com/PolicyEngine/policyengine-app/pull/543/files?diff=unified&w=0#diff-43f598ac8f8928c8be5adb376f0f5e8d25a5f61fb6dea523f1157c84d2517ed9L55-R59))
  - Add console log statements for debugging API results ([link](https://github.com/PolicyEngine/policyengine-app/pull/543/files?diff=unified&w=0#diff-43f598ac8f8928c8be5adb376f0f5e8d25a5f61fb6dea523f1157c84d2517ed9R71), [link](https://github.com/PolicyEngine/policyengine-app/pull/543/files?diff=unified&w=0#diff-43f598ac8f8928c8be5adb376f0f5e8d25a5f61fb6dea523f1157c84d2517ed9R86))
  - Extract error and loading states into separate components ([link](https://github.com/PolicyEngine/policyengine-app/pull/543/files?diff=unified&w=0#diff-43f598ac8f8928c8be5adb376f0f5e8d25a5f61fb6dea523f1157c84d2517ed9L91-R217))
  - Use `ResultsPanel` component to wrap title and description ([link](https://github.com/PolicyEngine/policyengine-app/pull/543/files?diff=unified&w=0#diff-43f598ac8f8928c8be5adb376f0f5e8d25a5f61fb6dea523f1157c84d2517ed9L91-R217))
  - Refactor plot data and layout into variables ([link](https://github.com/PolicyEngine/policyengine-app/pull/543/files?diff=unified&w=0#diff-43f598ac8f8928c8be5adb376f0f5e8d25a5f61fb6dea523f1157c84d2517ed9L91-R217))
  - Add radio group to toggle between baseline and reform rates or difference ([link](https://github.com/PolicyEngine/policyengine-app/pull/543/files?diff=unified&w=0#diff-43f598ac8f8928c8be5adb376f0f5e8d25a5f61fb6dea523f1157c84d2517ed9L91-R217))
  - Update colors and styles of plot to match app design ([link](https://github.com/PolicyEngine/policyengine-app/pull/543/files?diff=unified&w=0#diff-43f598ac8f8928c8be5adb376f0f5e8d25a5f61fb6dea523f1157c84d2517ed9L91-R217))
* Update and simplify `VariableEditor` component to improve performance and user experience ([link](https://github.com/PolicyEngine/policyengine-app/pull/543/files?diff=unified&w=0#diff-0b96b3d13d96221f5cb68df970385b96da6e895d2437c2fd666a3d052401b991L150-L151), [link](https://github.com/PolicyEngine/policyengine-app/pull/543/files?diff=unified&w=0#diff-0b96b3d13d96221f5cb68df970385b96da6e895d2437c2fd666a3d052401b991L179-R177), [link](https://github.com/PolicyEngine/policyengine-app/pull/543/files?diff=unified&w=0#diff-0b96b3d13d96221f5cb68df970385b96da6e895d2437c2fd666a3d052401b991L241-R245), [link](https://github.com/PolicyEngine/policyengine-app/pull/543/files?diff=unified&w=0#diff-0b96b3d13d96221f5cb68df970385b96da6e895d2437c2fd666a3d052401b991L253-R264))
  - Remove unused variable declaration ([link](https://github.com/PolicyEngine/policyengine-app/pull/543/files?diff=unified&w=0#diff-0b96b3d13d96221f5cb68df970385b96da6e895d2437c2fd666a3d052401b991L150-L151))
  - Replace `setSearchParams` with `window.history.replaceState` to avoid page reload ([link](https://github.com/PolicyEngine/policyengine-app/pull/543/files?diff=unified&w=0#diff-0b96b3d13d96221f5cb68df970385b96da6e895d2437c2fd666a3d052401b991L179-R177))
  - Update props passed to `SearchOptions` component to match metadata format and make component controlled ([link](https://github.com/PolicyEngine/policyengine-app/pull/543/files?diff=unified&w=0#diff-0b96b3d13d96221f5cb68df970385b96da6e895d2437c2fd666a3d052401b991L241-R245))
  - Simplify layout and styling of `HouseholdVariableEntityInput` component and display reform value ([link](https://github.com/PolicyEngine/policyengine-app/pull/543/files?diff=unified&w=0#diff-0b96b3d13d96221f5cb68df970385b96da6e895d2437c2fd666a3d052401b991L253-R264))

fixes #531

### CODE REVIEW

File: src/pages/household/input/VariableEditor.jsx [hunk 1]
```diff
@@ -176,7 +174,7 @@ function HouseholdVariableEntityInput(props) {
         (householdId) => {
           let newSearch = new URLSearchParams(window.location.search);
           newSearch.set("household", householdId);
-          setSearchParams(newSearch);
+          window.history.replaceState({}, '', `${window.location.pathname}?${newSearch}`);
```
Comment (nikhilwoodruff) [inline src/pages/household/input/VariableEditor.jsx:177]: Is there a reason for this? My understanding was that using `setSearchParams` ensured that React triggered all the processes needed to respond to a change in state. Does that still happen here?

File: src/pages/household/output/MarginalTaxRates.jsx [hunk 2]
```diff
@@ -17,12 +17,12 @@ import { ChartLogo } from "../../../api/charts";
 
 export default function MarginalTaxRates(props) {
   const { householdInput, householdBaseline, metadata, policyLabel, policy } = props;
-  const [baselineMtr, setBaselineMtr] = useState(null);
+  const [baselineMtr, setBaselineMtr] = useState([]);
```
Comment (nikhilwoodruff) [inline src/pages/household/output/MarginalTaxRates.jsx:20]: Ah OK this fix makes sense, thanks

**Review by nikhilwoodruff — COMMENTED** [review]
Thanks @sumitshinde-84 - left a few comments. Could you also add a screenshot of the fix in action?

**Comment by anth-volk** [comment]
Considering the age on this, I'm closing this out. I think so much has changed on the front end in this area that merging would be irresponsible.

**Pull request closed without merging**